### DISCUSSION TITLE
Debugger with decoupled pixel ratio

### DISCUSSION THREAD

**Discussion by Paril** [mikke89/RmlUi#638] [Q&A] [answered]
Hey there - I had a quick question about Rml::Debugger. So, the way it is currently, the debugger is attached to a context and renders within that context. This means that it seemingly inherits the pixel ratio of that context; we use a context that uses a DOS-sized scale and changes depending on a few things, meaning the debugger is often either too large or too small (you have to have a very specific size for it to be correct).

Is there any easy way currently to somehow decouple these ratios so that the debugger is always 1 px = 1 px, or separate them so we can allow it to be used in high DPI mode without being coupled to the real UIs' scale? Alternatively, could it be set up so that a debugger is attached to two contexts, one for what you're debugging and a second one for what you're rendering to?

**Comment by mikke89** [comment]
Hello!

> Alternatively, could it be set up so that a debugger is attached to two contexts, one for what you're debugging and a second one for what you're rendering to?

I believe we already have you covered here, you can do `Rml::Debugger::SetContext` to debug another context.

**Reply by Paril** [reply]
Hey;

Thanks for replying, I don't think that's what I want, though. Just as an example, let's say I have a UI displayed on a 1920x1080 display, with the pixel ratio set to (1080 / 240) because the original pixel-based UI was rendered at 320x240. If I use the Rml::Debugger on this context, the debugger is extremely large because it has built-in rcss that references `dp` units (`CommonSource.h`).

Is there a way around this? I don't think I can change the rcss for the debugger itself because it's isolated in a document I can't touch, and I don't think I can split the debugger into two pieces (a "debugger" context vs a "render the debugger of the other context to this context" context).

**Reply by mikke89** [reply, marked as answer]
> I don't think I can split the debugger into two pieces (a "debugger" context vs a "render the debugger of the other context to this context" context).

This is exactly what I mean here though, you can debug another context than the one your debugger is rendered in. So you just setup the context the debugger runs in with a "normal" dp-ratio, and then in the target context you can set it to something different.

**Reply by Paril** [reply]
Oh I see, okay. The API is just a bit confusing then because I thought it was changing the context being debugged, heh. Thanks, I'll give this a shot!
